server/src/main/java/io/crate/types/StringType.java

```diff
@@ -69,6 +70,18 @@ private StringType(int lengthLimit) {
         this.lengthLimit = lengthLimit;
     }
 
+    public StringType(StreamInput in) throws IOException {
+        if (in.getVersion().onOrAfter(Version.V_4_2_0)) {
+            if (in.readBoolean()) { // see writeTo
+                lengthLimit = Integer.MAX_VALUE;
+            } else {
+                lengthLimit = in.readInt();
```

Comment:
Given that we already use `readInt` /`writeInt` instead of the `vInt` variants, couldn't we just write the `MAX_VALUE` directly and avoid the boolean?